component/azstorage/block_blob.go

```diff
@@ -637,6 +646,35 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 				attr.Flags.Set(internal.PropFlagMetadataRetrieved)
 				attr.Flags.Set(internal.PropFlagModeDefault)
 				blobList = append(blobList, attr)
+
+			} else if bb.listDetails.Permissions {
```

Comment:
Please refactor the method in small small functionality.
Right now, it's becoming too big to understand.

Reply:
Made the changes please take a look